1172 implementors in the new transcript

SebastianLorenzano opened the pull request:
changed text variable from a SpTextPresenter to a SpCodePresenter and removed the syntaxHighlight and lineNumbers. This allows for things like for e.g  <meta> + p, etc.

fixed tests (they weren't always working due needing a wait to have the presenter updated outside of the main thread. This happens because of the structure of the new transcript: The new transcript has a main instance that sends entries via events, and the instances of the TranscriptPresenter recieve the event and update their text using it. Due to this, a wait is needed for the update to be done, and it needs to happen outside of the main thread.

'deleting extra takeKeyboardFocus' is a commit that I mistakenly put here but since it doesn't really matter because it just removes a dead line of code from the StPulse I will leave it here :)

https://github.com/pharo-spec/NewTools/issues/1172

SebastianLorenzano committed:
deleting extra takeKeyboardFocus

SebastianLorenzano committed:
changed text variable from a SpTextPresenter to a SpCodePresenter and removed the syntaxHighlight and lineNumbers. This allows for things like for e.g  <meta> + p, etc.

SebastianLorenzano committed:
fixed tests (they weren't always working due needing a wait to have the presenter updated outside of the main thread. This happens because of the structure of the new transcript: The new transcript has a main instance that sends entries via events, and the instances of the TranscriptPresenter recieve the event and update their text using it. Due to this, a wait is needed for the update to be done, and it needs to happen outside of the main thread

Ducasse commented on src/NewTools-Transcript-Tests/StTranscriptPresenterTest.class.st:
```diff
@@ -32,8 +33,9 @@ StTranscriptPresenterTest >> testAnnouncementIsWritten [
 		transcript show: 'test'.
-		300 milliSeconds wait.
-		self assert: presenter text text equals: 'test' ]
+		[ 
```
I do not know if this correct to use this here. 
@guillep what do you think?

SebastianLorenzano replied:
I had no other way to make the tests work without either making the wait async or halting when making the test. Is there any other way to go through the tests? The reason is the one explained in the commit :P

SebastianLorenzano replied:
changed it to use a backendForTest >> waitUntilUIRedrawed instead, like StPlaygroundTest and most others do :)

SebastianLorenzano committed:
changed tests from using forks to use their dedicated worker

SebastianLorenzano committed:
refactored text presenter name from 'text' to 'textPresenter' for clarity.
changed tests to work with waitUntilUIRedrawed (backendForTest waitUntilUIRedrawed) instead of using a worker.
changed protocol of layout to accessing.
